### PR TITLE
Ingress IP docs -- edits complete

### PR DESCRIPTION
**Pull request by jputrino**
@sjberman 

#### What issues does this address?
Fixes #319 

#### What's this change do?
- Edit Ingress default IP docs; fix up and clarify Ingress examples
- ensure consistency across examples -- partition name, schema version

#### Where should the reviewer start?
View the [staged documentation](https://s3-us-west-2.amazonaws.com/staging-c2ub89n2qjgt1/f5-ci-docs/pr328/kubernetes/kctlr-ingress.html).

#### Any background context?

Linkcheck will fail the build in travis -- contains a link to product docs for v1.4 that don't exist online yet.

### CODE REVIEW

File: docs/kubernetes/config_examples/f5-k8s-ingress-fanout.yaml [hunk 1]
```diff
@@ -4,22 +4,23 @@ metadata:
   name: ing-fanout
   namespace: default
   annotations:
-    # IP address of a BIG-IP pool member
+    # Provide an IP address from the external VLAN on your BIG-IP device;
+    # a hostname resolvable by DNS; or use the default IP
```
Comment (sjberman) [inline docs/kubernetes/config_examples/f5-k8s-ingress-fanout.yaml]: If users want to use the DNS lookup feature, the hostname that gets resolved is the first "host" under "spec:rules" below, not the annotation. (This applies to all comments that you have added in these Ingress examples.)

File: docs/kubernetes/kctlr-ingress.rst [hunk 2]
```diff
@@ -5,195 +5,187 @@
...
+If you're using multiple Controllers to monitor separate namespaces, you can define a default IP address for each Controller. This type of deployment allows you to isolate the VIPs in each namespace from each other.
 
-You can add the `supported Ingress annotations`_ to any existing Ingress resource using :command:`kubectl annotate` or :command:`oc annotate`.
-At minimum, define the following properties:
+DNS lookup
+~~~~~~~~~~
 
-- :code:`virtual-server.f5.com/ip`
-- :code:`virtual-server.f5.com/partition`
+The |kctlr| uses DNS lookup to resolve hostnames by default (as of v1.3.0). Provide the hostname in the ``virtual-server.f5.com/ip=`` annotation. The Controller will attempt to resolve the hostname and assign the resolved host's IP address to the Ingress' virtual server.
```
Comment (sjberman) [inline docs/kubernetes/kctlr-ingress.rst]: Same comment as above; the hostname comes from the first host in the Ingress spec, not the annotation.

File: docs/kubernetes/kctlr-ingress.rst [hunk 3]
```diff
@@ -233,45 +224,93 @@ Kubernetes Secrets
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A :dfn:`Name-based virtual hosting` ingress creates the following BIG-IP objects:
+
+- One (1) virtual server with one (1) pool for each Service.
+- Local traffic policies that route requests to specific pools based on host name and path.
+
+.. tip::
+
+   If you don't specify any hosts or paths, the BIG-IP device will proxy traffic for all hosts/paths for the Service specified in the :code:`backend` section of the virtual-server annotation.
```
Comment (sjberman) [inline docs/kubernetes/kctlr-ingress.rst]: I don't think virtual-server annotation is the right term here. I think it should be Ingress spec, or Ingress definition.

File: docs/kubernetes/kctlr-ingress.rst [hunk 4]
```diff
@@ -5,195 +5,187 @@
...
+If you're using multiple Controllers to monitor separate namespaces, you can define a default IP address for each Controller. This type of deployment allows you to isolate the VIPs in each namespace from each other.
 
-You can add the `supported Ingress annotations`_ to any existing Ingress resource using :command:`kubectl annotate` or :command:`oc annotate`.
-At minimum, define the following properties:
+DNS lookup
+~~~~~~~~~~
 
-- :code:`virtual-server.f5.com/ip`
-- :code:`virtual-server.f5.com/partition`
+The |kctlr| uses DNS lookup to resolve hostnames by default (as of v1.3.0). When you provide the hostname in the :code:`spec` section of the k8s-bigip-ctlr Deployment file, the Controller attempts to resolve that hostname. It then assigns the resolved host's IP address to the Ingress' virtual server.
```
Comment (sjberman) [inline docs/kubernetes/kctlr-ingress.rst]: It's actually the 'spec' section in the Ingress resource, not the k8s Deployment. There can be multiple hosts provided in an Ingress 'spec', we use the first host we see and attempt to resolve it.

Comment (jputrino) [reply docs/kubernetes/kctlr-ingress.rst]: Fixed!

**Pull request merged**